Source/MonoGame.Extended/Collections/ObjectPool.Diagnostics.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System.Threading;
+
+namespace MonoGame.Extended.Collections
+{
+    public partial class ObjectPool
+    {
+        public class Diagnostics
+        {
```

Comment:
It's from the derived code, http://www.codeproject.com/Articles/535735/Implementing-a-Generic-Object-Pool-in-NET.
It's mostly useful for unit testing, but I can for see it being useful in a UI widget for getting a quick snapshot of the state of the pool.